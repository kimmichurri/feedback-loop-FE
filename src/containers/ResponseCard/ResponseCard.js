import React, { Component } from 'react'

export class ResponseCard extends Component {
  constructor() {
    super();
    this.state = {
      response: {}
    }
  }
  
  handleChange = (e) => {
    this.setState({
      response: {
        question: this.props.question.id,
        currStudent: this.props.currStudent,
        member: this.props.member.id, 
        pointValue: parseInt(e.target.value)
      }
    }, () => {this.props.checkResponse(this.state.response)})
  }

  render() {
    return (
      <form className='response-card'>
        <p className='question-title'>{this.props.question.questionTitle}</p>
        <div className='response-container'>
          <div className='response-option'>
            <input 
              type="radio" 
              name="option" 
              value={this.props.question.options[0].pointValue}
              onChange={this.handleChange}
            />
            <p className='response-description'>{this.props.question.options[0].description}</p>
          </div>
          <div className='response-option'>
            <input 
              type="radio" 
              name="option"
              value={this.props.question.options[1].pointValue}
              onChange={this.handleChange}
            />
            <p className='response-description'>{this.props.question.options[1].description}</p>
          </div>
          <div className='response-option'>
            <input 
              type="radio" 
              name="option"
              value={this.props.question.options[2].pointValue}
              onChange={this.handleChange}
            />
            <p className='response-description'>{this.props.question.options[2].description}</p>
          </div>
          <div className='response-option'>
            <input 
              type="radio" 
              name="option"
              value={this.props.question.options[3].pointValue}
              onChange={this.handleChange}
            />
            <p className='response-description'>{this.props.question.options[3].description}</p>
          </div>
        </div>
      </form>
    )
  }
}

export default ResponseCard