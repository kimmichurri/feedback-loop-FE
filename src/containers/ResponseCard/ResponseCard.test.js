import React from 'react'
import { ResponseCard } from './ResponseCard'
import { shallow } from 'enzyme'

describe('ResponseCard', () => {
  let wrapper
  let mockQuestion = { 
    id: 7,
    questionTitle: 'My Question',
    options: [{
        id: 1,
        description: "Option One"
      },
      {
        id: 2,
        description: "Option Two"
      },
      {
        id: 3,
        description: "Option Three"
      },
      {
        id: 4,
        description: "Option Four"
      },
    ]
  }
  let mockMember = {
    id: 13
  }
  let mockFn = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<ResponseCard 
      question={mockQuestion}
      member={mockMember}
      checkResponse={mockFn}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
  it('should have default state', () => {
    const expected = {
      response: {}
    }
    expect(wrapper.state()).toEqual(expected)
  })

  it('should update state onChange', () => {
    const mockEvent = { target: { name: 'option1', value: 1 }}
    const expected = {
      response: {
        question: 7,
        member: 13,
        answer: 1
      }
    }
    wrapper.find('#op1').simulate('change', mockEvent)
    expect(wrapper.state()).toEqual(expected)
  })

  it('should call checkResponse when handleChange is called', () => {
    const mockEvent = { target: { name: 'option1', value: 1 }}
    wrapper.instance().handleChange(mockEvent)
    expect(mockFn).toHaveBeenCalled()
  })
})