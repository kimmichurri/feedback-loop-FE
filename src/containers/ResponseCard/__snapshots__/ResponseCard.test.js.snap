// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponseCard should match the snapshot 1`] = `
<form
  className="response-card"
>
  <p
    className="question-title"
  >
    My Question
  </p>
  <div
    className="response-container"
  >
    <div
      className="response-option"
    >
      <input
        id="op1"
        name="option"
        onChange={[Function]}
        type="radio"
        value={1}
      />
      <p
        className="response-description"
      >
        Option One
      </p>
    </div>
    <div
      className="response-option"
    >
      <input
        name="option"
        onChange={[Function]}
        type="radio"
        value={2}
      />
      <p
        className="response-description"
      >
        Option Two
      </p>
    </div>
    <div
      className="response-option"
    >
      <input
        name="option"
        onChange={[Function]}
        type="radio"
        value={3}
      />
      <p
        className="response-description"
      >
        Option Three
      </p>
    </div>
    <div
      className="response-option"
    >
      <input
        name="option"
        onChange={[Function]}
        type="radio"
        value={4}
      />
      <p
        className="response-description"
      >
        Option Four
      </p>
    </div>
  </div>
</form>
`;
